test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,204 @@
+use futures::stream::Stream;
+use futures::StreamExt;
+// use opentelemetry::global::shutdown_tracer_provider;
+use opentelemetry::sdk::metrics::{selectors, PushController};
+// use opentelemetry::trace::TraceError;
+use opentelemetry::global;
+use opentelemetry::metrics::{self, ObserverResult, ValueRecorder};
+use opentelemetry_otlp::ExporterConfig;
+use std::error::Error;
+use std::time::Duration;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+
+use clap::{value_t, App, Arg};
+use rand::Rng;
+
+// Skip first immediate tick from tokio, not needed for async_std.
+fn delayed_interval(duration: Duration) -> impl Stream<Item = tokio::time::Instant> {
+    opentelemetry::util::tokio_interval_stream(duration).skip(1)
+}
+
+fn init_meter(period: f64, otlp_endpoint: String) -> metrics::Result<PushController> {
+    let export_config = ExporterConfig {
+        endpoint: otlp_endpoint.to_string(),
+        ..ExporterConfig::default()
+    };
+    opentelemetry_otlp::new_metrics_pipeline(tokio::spawn, delayed_interval)
+        .with_export_config(export_config)
+        .with_aggregator_selector(selectors::simple::Selector::Exact)
+        .with_period(Duration::from_secs_f64(period))
+        .build()
+}
+
+#[derive(Debug)]
+struct Config {
+    update_rate: f64,
+    push_rate: f64,
+    batching_enabled: bool,
+    otlp_endpoint: String,
+    enable_value_recorder: bool,
+    enable_value_observer: bool,
+}
+
+fn init_config() -> Result<Config, Box<dyn Error + Send + Sync + 'static>> {
+    let matches = App::new("obs_agent_client")
+        .arg(
+            Arg::with_name("update-rate")
+                .short("u")
+                .long("update-rate")
+                .help("Rate at which each instrument is updated with a new metric measurement (updates/sec)")
+        )
+        .arg(
+            Arg::with_name("push-rate")
+                .short("p")
+                .long("push-rate")
+                .help("Rate at which measurements are pushed out of the client (pushes/sec)")
+        )
+        .arg(
+            Arg::with_name("batching-enabled")
+                .short("b")
+                .long("batching-enabled")
+                .help("Enables or disables batch recording of measurements.")
+        )
+        .arg(
+            Arg::with_name("otlp-endpoint")
+                .short("e")
+                .long("otlp-endpoint")
+                .help("Endpoint to which OTLP messages will be sent.")
+        )
+        .arg(
+            Arg::with_name("enable-value-recorder")
+                .short("r")
+                .long("enable-value-recorder")
+                .help("Enables or disables value recorder instrument.")
+        )
+        .arg(
+            Arg::with_name("enable-value-observer")
+                .short("v")
+                .long("enable-value-observer")
+                .help("Enables or disables value observer instrument.")
+        )
+        .get_matches();
```

Comment:
So we are providing configuration via arguments or environment variables, where env vars take precedence? This seems ok to me. The general pattern we follow for modules like this is to make a settings file like this (modeled after the bridge):
https://github.com/Azure/iotedge/blob/master/test/modules/generic-mqtt-tester/src/settings.rs

Is the point of the arguments to make it easier to run from the terminal?

Reply:
Exactly - there are scenarios where it is convenient to run this program from the cmd line which is why I decided to use clap, but for configuration when deployed as an Edge module I have the env vars since we can't yet use desired properties with Rust modules.

I like the json approach too. We can certainly switch to that pattern in the future if we want consistency.

Reply:
I did find a package `clap_conf` which will combine env vars with command line arguments, but it doesn't incorporate a json file.

Reply:
I took a look at `clap_conf` but I'm reluctant to use it right now because it's only on version 0.1.5 and hasn't had any activity in it's git repo in almost a year.